consensus/p2p/statemachine/transition.go

```diff
@@ -94,68 +97,95 @@ func (t *transition) OnEmptyBlockCommitment(
 	}, nil
 }
 
-// TODO: Implement this function properly
-func (t *transition) OnBlockInfo(
+func (t *transition[V, H, A]) OnBlockInfo(
 	ctx context.Context,
 	state *AwaitingBlockInfoOrCommitmentState,
 	blockInfo *consensus.BlockInfo,
 ) (*ReceivingTransactionsState, error) {
+	adaptedBlockInfo, err := p2p2consensus.AdaptBlockInfo(blockInfo)
+	if err != nil {
+		return nil, err
+	}
+	t.validator.BlockInfo(&adaptedBlockInfo)
+
 	return &ReceivingTransactionsState{
 		Header:     state.Header,
 		ValidRound: state.ValidRound,
-		Value:      utils.HeapPtr(starknet.Value(felt.FromUint64(blockInfo.Timestamp))),
+		// Todo: remove, OnBlockInfo can't know value / block hash yet..
+		Value: utils.HeapPtr(starknet.Value(felt.FromUint64(blockInfo.Timestamp))),
 	}, nil
 }
 
-// TODO: Implement this function properly
-func (t *transition) OnTransactions(
+func (t *transition[V, H, A]) OnTransactions(
 	ctx context.Context,
 	state *ReceivingTransactionsState,
 	transactions []*consensus.ConsensusTransaction,
 ) (*ReceivingTransactionsState, error) {
+	txns := make([]types.Transaction, len(transactions))
+	for i := range transactions {
+		txn, class, err := p2p2consensus.AdaptTransaction(transactions[i])
+		if err != nil {
+			return nil, err
+		}
+		txns[i] = types.Transaction{
+			Transaction: txn,
+			Class:       class,
+		}
+	}
+
+	err := t.validator.TransactionBatch(txns)
+	if err != nil {
+		return nil, err
+	}
+
 	return &ReceivingTransactionsState{
 		Header:     state.Header,
 		ValidRound: state.ValidRound,
-		Value:      state.Value,
+		// Todo: remove, OnTransactions can't know value / block hash yet..
+		Value: state.Value,
 	}, nil
 }
 
-// TODO: Implement this function properly
-func (t *transition) OnProposalCommitment(
+func (t *transition[V, H, A]) OnProposalCommitment(
 	ctx context.Context,
 	state *ReceivingTransactionsState,
 	commitment *consensus.ProposalCommitment,
 ) (*AwaitingProposalFinState, error) {
+	adaptedCommitment, err := p2p2consensus.AdaptProposalCommitment(commitment)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = t.validator.ProposalCommitment(&adaptedCommitment); err != nil {
+		return nil, err
+	}
+
 	return &AwaitingProposalFinState{
 		Header:     state.Header,
 		ValidRound: state.ValidRound,
-		Value:      state.Value,
+		// Todo: remove, we aren't populating with the value/block hash we generate..
+		Value: state.Value,
```

Comment:
We should populate it here, because at this point, we already have the full `ProposalCommitment`, so we can calculate the hash. When `ProposalFin` arrives, what we need is to compare the calculated hash and the received hash.